src/migration/1714633817175-SquadTotalPostsTrigger.ts

```diff
@@ -0,0 +1,67 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class SquadTotalPostsTrigger1714633817175 implements MigrationInterface {
+  name = 'SquadTotalPostsTrigger1714633817175';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    queryRunner.query(`
+        CREATE OR REPLACE FUNCTION increment_squad_posts_count()
+          RETURNS TRIGGER
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        DECLARE
+          s public.source;
+        BEGIN
+          SELECT * INTO s FROM public.source WHERE id = NEW."sourceId";
+          IF s.type = 'squad' THEN
+            UPDATE source
+            SET flags = jsonb_set(flags, array['totalPosts'], to_jsonb(COALESCE(CAST(flags->>'totalPosts' AS INTEGER), 0) + 1))
```

Comment:
Never seen this `array['totalPosts']` syntax. It actually returns `{totalPosts}` which is what `jsonb_set` requires. I would prefer to just use string, seems more simple if there is not anything else I am missing?

Reply:
The `array['totalPosts']` just looks more object-oriented and modern, I have no hard preference, so either way works for me.